include/exec/async_scope.hpp

```diff
@@ -21,6 +21,7 @@
 #include "env.hpp"
 
 #include <mutex>
+#include <optional>
```

Comment:
could you use `<stdexec/__detail/__optional.hpp>` instead? or even `<stdexec/__detail/__manual_lifetime.hpp>`? `std::optional` is a bit of a pig at compile-time.